[IMP] IoT: edits to all IoT docs (#3673)
odoo/documentation: content/applications/productivity/iot/devices/printer.rst

=================
Connect a printer
=================

A printer can be attached to the :abbr:`IoT (Internet of Things)` box on an Odoo database.

> Installation is easy and convenient and can be done in a few easy steps. The printer can be used to
> print receipts, labels, orders or even reports from the different Odoo apps. In addition printer
> actions can be assigned as an action on a trigger during the manufacturing process or added onto a
> *Quality Control Point* or *Quality Check*.

Reviewer: ```suggestion
Installation is easy, convenient and can be done in a few easy steps. The printer can be used to
print receipts, labels, orders, or even reports from the different Odoo apps. In addition, printer
actions can be assigned as an *action on a trigger* during the manufacturing process, or added onto
a quality control point or a quality check.
```

Connection
==========

The :abbr:`IoT (Internet of Things)` box supports printers connected through :abbr:`USB (Universal

> Serial Bus)`, network connection or Bluetooth. `Supported printers

Reviewer: ```suggestion
Serial Bus)`, network connection, or Bluetooth. `Supported printers
```

> <https://www.odoo.com/page/iot-hardware>`__ will be detected automatically and will appear in the
> :guilabel:`Devices` list of the :menuselection:`IoT` app.

Reviewer: ```suggestion
<https://www.odoo.com/page/iot-hardware>`__ are detected automatically, and appear in the
:guilabel:`Devices` list of the :menuselection:`IoT` app.
```

.. image:: printer/printer-detected.png
   :align: center
   :alt: The printer as it would appear in the IoT app devices list.

.. note::
   The printer can take up to two minutes to appear in the :menuselection:`IoT` app devices list.

> Link the printer
> ================

Reviewer: ```suggestion
Link printer
============
```

Link printer to work orders
---------------------------

> *Work Orders* can be linked to printers via a *Quality Control Point* to print labels for
> manufactured products.

Reviewer: ```suggestion
*Work Orders* can be linked to printers via a quality control point to print labels for
manufactured products.
```

> In the :menuselection:`Quality` app, a device can be set up on a :guilabel:`Quality Control Point`.
> Go to the :menuselection:`Quality Control --> Control Points` and open the :guilabel:`Control Point`
> which will be linked with the printer.

Reviewer: ```suggestion
In the *Quality* app, a device can be set up on a quality control point. To do that, go to the
:menuselection:`Quality app --> Quality Control --> Control Points`, and open the desired control
point to which the printer will be linked.
```

.. note::

>    A *Manufacturing Operation* and *Work Order Operation* need to be attached to the *Quality
>    Control Point* before the :guilabel:`Type` will allow for :guilabel:`Print Label` to be
>    selected.

Reviewer: ```suggestion
   A *Manufacturing Operation* and *Work Order Operation* need to be attached to a quality control
   point before the :guilabel:`Type` field allows for the :guilabel:`Print Label` option to be
   selected.
```

> Now, edit the :guilabel:`Control Point` by selecting the :guilabel:`Type` field and clicking on
> :guilabel:`Print Label` from the dropdown. A field called :guilabel:`Device` will appear where
> the attached *device* can be selected. :guilabel:`Save` the changes if required.

Reviewer: ```suggestion
From here, edit the control point, by selecting the :guilabel:`Type` field, and selecting
:guilabel:`Print Label` from the drop-down menu of options. Doing so reveals a field called
:guilabel:`Device`, where the attached device can be selected. :guilabel:`Save` the changes, if
required.
```

> Now, edit the :guilabel:`Control Point` by selecting the :guilabel:`Type` field and clicking on
> :guilabel:`Print Label` from the dropdown. A field called :guilabel:`Device` will appear where
> the attached *device* can be selected. :guilabel:`Save` the changes if required.

Reviewer: ```suggestion
From here, edit the control point, by selecting the :guilabel:`Type` field, and selecting
:guilabel:`Print Label` from the drop-down menu of options. Doing so reveals a field called
:guilabel:`Device`, where the attached device can be selected. :guilabel:`Save` the changes, if
required.
```

.. image:: printer/printer-controlpoint.png
   :align: center
   :alt: This is the quality control point setup.

> The printer can be used with the selected :guilabel:`Quality Control Point`. When the
> :guilabel:`Quality Control Point` is reached during the manufacturing process, the database will
> prompt the operator to take a picture.

Reviewer: ```suggestion
The printer can be used with the selected quality control point. When the quality control point is
reached during the manufacturing process, the database prompts the operator to take a picture.
```

> The printer can be used with the selected :guilabel:`Quality Control Point`. When the
> :guilabel:`Quality Control Point` is reached during the manufacturing process, the database will
> prompt the operator to take a picture.

Reviewer: ```suggestion
The printer can be used with the selected quality control point. When the quality control point is
reached during the manufacturing process, the database prompts the operator to take a picture.
```

.. image:: printer/printer-prompt.png
   :align: center

.. seealso::

>    In a *Quality Check* the :guilabel:`Type` of check can also be specified to :guilabel:`Print
>    Label`. Access *Quality Checks* by navigating to :menuselection:`Quality app --> Quality Control
>    --> Quality Checks --> New`.

Reviewer: ```suggestion
   On a quality check detail form, the :guilabel:`Type` of check can also be specified to
   :guilabel:`Print Label`. To create new quality checks, navigate to :menuselection:`Quality app
   --> Quality Control --> Quality Checks --> New`.
```

.. seealso::
   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_control_points`
   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_alerts`

> Link a printer to a work center in the manufacturing app
> --------------------------------------------------------

Reviewer: ```suggestion
Link printer to work center in manufacturing app
------------------------------------------------
```

> To link the printer to an action, it needs to be configured on a *work center*. Navigate to
> :menuselection:`Manufacturing --> Configuration --> Work Centers`. Go to the :guilabel:`Work Center`
> the printer will be used in and add the device in the :guilabel:`IoT Triggers` tab under
> :guilabel:`Device` by selecting :guilabel:`Add a Line`. Then, it can be linked to either of the
> following :guilabel:`Actions`: :guilabel:`Print Labels`, :guilabel:`Print Operation`, or
> :guilabel:`Print Delivery Slip`. A key can be added to trigger the action.

Reviewer: ```suggestion
To link a printer to an action, it first needs to be configured on a work center. To do that,
navigate to :menuselection:`Manufacturing app --> Configuration --> Work Centers`. From here,
select the desired work center in which the printer will be used. Next, add the device in the
:guilabel:`IoT Triggers` tab, under the :guilabel:`Device` column, by selecting
:guilabel:`Add a Line`.

Then, the printer can be linked to either of the following options in the :guilabel:`Actions`
drop-down menu: :guilabel:`Print Labels`, :guilabel:`Print Operation`, or :guilabel:`Print
Delivery Slip`. A key can be also added to trigger the action.
```

> It should be noted that the trigger that is first in the list will be chosen first. So, the order
> matters and these triggers can be dragged into order.

Reviewer: ```suggestion
.. important::
   It should be noted that the first listed trigger is chosen first. So, the order matters, and
   these triggers can be dragged into any order.
```

.. note::

>    On the :guilabel:`work order` screen, a status graphic indicates whether the database is
>    correctly connected to the printer.

Reviewer: ```suggestion
   On the :guilabel:`Work Order` screen, a status graphic indicates whether the database is
   correctly connected to the printer.
```

>    On the :guilabel:`work order` screen, a status graphic indicates whether the database is
>    correctly connected to the printer.

Reviewer: ```suggestion
   On the :guilabel:`Work Order` screen, a status graphic indicates whether the database is
   correctly connected to the printer.
```

.. seealso::
   :ref:`workcenter_iot`

> Link the printer to reports
> ---------------------------

Reviewer: ```suggestion
Link printer to reports
-----------------------
```

It's also possible to link a type of report to a certain printer. In the :menuselection:`IoT` app,

> go to the :menuselection:`Devices` menu and select the printer that needs to be configured.

Reviewer: ```suggestion
go to the :guilabel:`Devices` menu, and select the desired printer that needs to be configured.
```

> Now, go to the :guilabel:`Printer Reports` tab. :guilabel:`Edit` the page and select :guilabel:`Add
> a line`. In the window that appears, check all the types of :guilabel:`Reports` that should be
> linked to this printer.

Reviewer: ```suggestion
From here, click :guilabel:`Edit`, go to the :guilabel:`Printer Reports` tab, and select :guilabel:`Add
a line`. In the window that appears, check all the types of :guilabel:`Reports` that should be
linked to this printer.
```

.. image:: printer/printers-listed.png
   :align: center
   :alt: The printer devices listed in the IoT Devices menu.

Now, each time :guilabel:`Print` is selected in the control panel, instead of downloading a PDF,

> Odoo will send the report to the selected printer and automatically print it.

Reviewer: ```suggestion
Odoo sends the report to the selected printer, and automatically prints it.
```

.. seealso::
   :doc:`POS Order Printing <../../../sales/point_of_sale/restaurant/kitchen_printing>`

.. note::
   Reports can also be configured in the :guilabel:`Technical Menu` while in :ref:`debug mode

>    <developer-mode>`. Navigate to :menuselection:`Settings App --> Technical Menu --> Actions -->
>    Reports`. The individual report can be found in this list, where the :guilabel:`IOT Device` can
>    be set on the report.

Reviewer: ```suggestion
   <developer-mode>`. To do that, navigate to :menuselection:`Settings App --> Technical Menu -->
   Actions --> Reports`. From here, the individual report can be found in this list, where the
   :guilabel:`IoT Device` can be set on the report.
```

   can be added for the device.

.. seealso::
   On a quality check detail form, the :guilabel:`Type` of check can also be specified to
   :guilabel:`Print Label`. To create new quality checks, navigate to :menuselection:`Quality app
   --> Quality Control --> Quality Checks --> New`.

.. seealso::
   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_control_points`
   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_alerts`

Link printer to work center in manufacturing app
------------------------------------------------

To link a printer to an action, it first needs to be configured on a work center. To do that,
navigate to :menuselection:`Manufacturing app --> Configuration --> Work Centers`. From here, select
the desired work center in which the printer will be used. Next, add the device in the
:guilabel:`IoT Triggers` tab, under the :guilabel:`Device` column, by selecting :guilabel:`Add a
Line`.

Then, the printer can be linked to either of the following options in the :guilabel:`Actions`
drop-down menu: :guilabel:`Print Labels`, :guilabel:`Print Operation`, or :guilabel:`Print Delivery

> Slip`. A key can be also added to trigger the action.

Reviewer: ```suggestion
Slip`. A key can also be added to trigger the action.
```

   these triggers can be dragged into any order.

.. note::
   On the :guilabel:`Work Order` screen, a status graphic indicates whether the database is
   correctly connected to the printer.

.. seealso::
   :ref:`workcenter_iot`

Link printer to reports
-----------------------

It's also possible to link a type of report to a certain printer. In the *IoT app*, go to the
:guilabel:`Devices` menu, and select the desired printer that needs to be configured.

From here, click :guilabel:`Edit`, go to the :guilabel:`Printer Reports` tab, and select
:guilabel:`Add a line`. In the window that appears, check all the types of :guilabel:`Reports` that
should be linked to this printer.

.. image:: printer/printers-listed.png
   :align: center
   :alt: The printer devices listed in the IoT Devices menu.

Now, each time :guilabel:`Print` is selected in the control panel, instead of downloading a PDF,
Odoo sends the report to the selected printer, and automatically prints it.

.. seealso::
   :doc:`POS Order Printing <../../../sales/point_of_sale/restaurant/kitchen_printing>`

.. note::
   Reports can also be configured in the :guilabel:`Technical Menu` while in :ref:`debug mode

>    <developer-mode>`. To do that, navigate to :menuselection:`Settings App --> Technical Menu -->

Reviewer 2: ```suggestion
   <developer-mode>`. To do that, navigate to :menuselection:`Settings app --> Technical Menu -->
```

=================
Connect a printer
=================

A printer can be attached to the *IoT Box* in a Odoo database. Installation is easy and convenient
and can be done in a few easy steps. The printer can used to print receipts, orders or even reports
from the different Odoo apps.

Connection
==========

The IoT Box supports printers connected through USB, network or Bluetooth. `Supported printers
<https://www.odoo.com/page/iot-hardware>`__ will be detected automatically and will appear in the
:guilabel:`Devices` list of the IoT app.

.. note::
         The printer can take up to two minutes to appear in the IoT app devices list.

.. image:: printer/printer_01.png
   :align: center
   :alt: The printer as it would appear in the IoT app devices list.

Link the printer
================

To work orders
--------------

*Work Orders* can be linked to printers via a *Quality Control Point* to print labels for
manufactured products.

> To do so, a *Quality Control Point* needs to be created within the *Quality* app. Afterwards the

Reviewer 3: ```suggestion
To do so, a *Quality Control Point* needs to be created within the :menuselection:`Quality` app. Once created, the
```

correct manufacturing operation and work order operation can be selected. In type, choose
:guilabel:`Print Label`. The changes will automatically save once navigating away from the page the
changes were made on.

.. image:: printer/printer_03.png
   :align: center
   :alt: This is the quality control point setup.

> Now, each time the quality control point is reached for the chosen product, a *Print Label* button

Author: Yes it's a button/icon.

> Now, each time the quality control point is reached for the chosen product, a *Print Label* button

Author: Thanks for catching this 👍

will appear.

.. image:: printer/printer_04.png
   :align: center
   :alt: The print label icon that populates once the quality control point is reached.

To reports
----------

It's also possible to link a type of report to a certain printer. In the :menuselection:`IoT` app,
go to the :guilabel:`Devices` menu and select the printer that needs to be configured.

.. image:: printer/printer_05.png
   :align: center
   :alt: The printer devices listed in the IoT Devices menu.

Now, go to the *Printer Reports* tab.

.. image:: printer/printer_06.png
   :align: center
   :alt: Printer configuration on the IoT devices list.

> Click edit and then, select :guilabel:`Add a line`. In the window that appears, check all the types

Reviewer 3: ```suggestion
Click :guilabel:`Edit`, then select :guilabel:`Add a line`. In the window that appears, check all the types
```

of reports that should be linked to this printer.

.. image:: printer/printer_07.png
   :align: center
   :alt: Report types to be added onto the printer configuration.

Now, each time :guilabel:`Print` is selected in the control panel, instead of downloading a PDF,
Odoo will send the report to the selected printer and automatically print it.

.. _iot/pos_printer:

Print receipts from the PoS
---------------------------

A printer can be linked to the :menuselection:`Point of Sale` app so that receipts can be printed
directly from the :guilabel:`PoS`.

Set up is simple and can be completed in just a few easy steps. Go to the
:menuselection:`Point of Sale` app and open the :guilabel:`Point of Sale` settings, under
:guilabel:`Configuration`. Select the :guilabel:`IoT Box` feature. Then, choose the
:guilabel:`Receipt Printer` from the dropdown. Save the settings.

.. image:: printer/printer_08.png
   :align: center
   :alt: IoT box configuration in the Point of Sale app.

> Now, different kinds of tickets can be printed from the :guilabel:`PoS` including **receipts**,

Reviewer 3: ```suggestion
Now, different kinds of tickets can be printed from the :guilabel:`PoS`. These include **receipts**,
```

**sale details** and **bills**.

Receipts are printed once the order is validated. The process is automated when the
feature is enabled in the :guilabel:`PoS configuration`.

Sales details can be printed by clicking on the printer icon on the navbar at the top of the
:guilabel:`PoS`. It will print the details of the sales of the current day.

.. image:: printer/printer_09.png
   :align: center
   :alt: Printer icon in the navbar of the POS.

As for the bill, it is only available in restaurant mode. In the restaurant settings, activate
:guilabel:`Print Bills` and a :guilabel:`Bill` button will appear in the left panel of the
:guilabel:`PoS`.

.. image:: printer/printer_10.png
   :align: center
   :alt: This is the Bill button as it appears in restaurant mode.

Print orders in the kitchen
---------------------------

In restaurant mode, order tickets can be sent to the kitchen.

To do so, go to the :menuselection:`PoS` app and open the :guilabel:`PoS` settings. Then, tick
:guilabel:`Order Printer`.

.. image:: printer/printer_11.png
   :align: center
   :alt: Order printer configuration in the POS settings.

> Now, go to the :guilabel:`Printers` menu. click on create, select the printer from the dropdown and,

Reviewer 3: ```suggestion
Then, go to the :guilabel:`Printers` menu. Click on create, select the printer from the dropdown, and
```
